src/classes/PMT_PaymentCreator.cls

```diff
@@ -402,15 +425,14 @@ public class PMT_PaymentCreator {
 
         // If the org has multicurrency set up we need to round the currency value
         if (UTIL_Currency.getInstance().isMultiCurrencyOrganization()) {
-            String oppCurrency = (String)newOpp.get('CurrencyIsoCode');
+            String oppCurrency = (UserInfo.isMultiCurrencyOrganization()) ? (String) newOpp.get('CurrencyIsoCode'): 'USD';
```

Comment:
I add the extra condition or a lot of test functions related to Advanced Currency Management will fail. The reason the test functions are failing because the test class is setting UTIL_Currency's multi-currency to true, but the org itself does not actually have the multi-currency and ACM enabled. Therefore, the roll-up on opp will still fire in test class and cause the opp stage to be evaluated. (The tests were not failing before because auto close opp stage evaluation was in the auto-payment creation condition and now it is not)

However, it's a good point not all org's default currency is USD. I update the hardcoded USD to UserInfo.getDefaultCurrency().